### ISSUE TITLE
Azure Container Instance (aci) State not returned

### ISSUE THREAD

**Issue by jsturtevant** [Azure/azure-libraries-for-net#118]
When listing Container Groups By Resource Group, the state of a Azure container instance the State property returns `null`:

```csharp
var sp = new ServicePrincipalLoginInformation
            {
                ClientId = "<ClientId>",
                ClientSecret = "<ClientSecret>"
            };
var credentials = new AzureCredentials(sp, "<tenantId>", AzureEnvironment.AzureGlobalCloud);
var azure = Azure.Authenticate(credentials);

var containerGroups = await azure.WithSubscription("<subid>")
                            .ContainerGroups
                            .ListByResourceGroupAsync("resourcegroup1");

var containerGroup = containerGroups.FirstOrDefault(x => x.Name == "acigroupname");
containerGroup.State == null
```

Whereas when you get by resource group it returns the correct value:

```csharp
var containerGroup = await azure.WithSubscription("<subid>")
                            .ContainerGroups.GetByResourceGroupAsync("resourcegroup1", "acigroupname");
containerGroup.State == "Running"
```

**Comment by milismsft** [comment]
I can confirm the issue and it seems to be a due to a bug on the service side rather than on the SDK side (the REST call return body is missing the "state" value). I'll follow up with the service owners on this issue.

**Comment by milismsft** [comment]
I was able to work around this issue through the Fluent implementation; the fix will be shipped as part of the v1.7.
